java/vector/src/main/codegen/templates/AbstractPromotableFieldWriter.java

```diff
@@ -42,6 +43,8 @@
    * @param type the type of the values we want to write
    * @return the corresponding field writer
    */
+  abstract protected FieldWriter getWriter(ArrowType type);
+
   abstract protected FieldWriter getWriter(MinorType type);
```

Comment:
Is is possible to just have `getWriter(ArrowType)`?  Having 2 methods so similar is a little confusing.. This is because you need a writer for a specific time unit now?

Reply:
I agree ideally we should have just one. However currently, this is needed to support this:

```
TimestampWriter timestampWriter = rootWriter.timestamp("a", TimeUnit.SECOND, "America/New_York");
timestampWriter.writeTimestamp(1000)
```

The second line here needs to lookup the writer by `MinorType.TIMESTAMP`

We need to do more refactoring to support this API without `abstract protected FieldWriter getWriter(MinorType type)`, but since this is going to be internal change and this PR is already quite complicated, I prefer this to be a follow up.

Reply:
Ok, sounds good